src/Plugin/Action/FetchEntityById.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\FetchEntityById.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\rules\Engine\RulesActionBase;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a 'Fetch entity by id' action.
+ *
+ * @Action(
+ *   id = "rules_fetch_entity_by_id",
+ *   label = @Translation("Fetch entity by id"),
+ *   category = @Translation("Entity"),
+ *   context = {
+ *     "entity_type" = @ContextDefinition("string",
+ *       label = @Translation("Entity type"),
+ *       description = @Translation("Specifies the type of entity that should be fetched.")
+ *     ),
+ *     "entity_id" = @ContextDefinition("integer",
+ *       label = @Translation("Identifier"),
+ *       description = @Translation("The id of the entity that should be fetched.")
+ *     )
+ *   },
+ *   provides = {
+ *     "entity" = @ContextDefinition("entity",
+ *       label = @Translation("Entity")
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ */
+class FetchEntityById extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The entity manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * Constructs a CreatePathAlias object.
```

Comment:
@diarmy: Small issue in the comment for the FetchEntityById Constructor: it currently says 'Constructs a CreatePathAlias object', rather than 'Constructs a FetchEntityById object'.